File: backend/app/routers/usage_analytics.py
# [AI GENERATED]
# Model: GitHub Copilot (GPT-4.1)
# Logic: Provides a backend FastAPI router for data usage analytics (dataset usage, report popularity, user engagement).
# Why: Enables tracking and reporting of analytics usage for business insights and optimization.
# Root Cause: No API endpoint for usage analytics existed.
# Context: Extend with real DB queries, filtering, and user-specific analytics as needed.
# Model Suitability: For advanced analytics, consider GPT-4 Turbo or Claude Sonnet.

from fastapi import APIRouter, Query
from typing import List, Dict

router = APIRouter()

@router.get("/analytics/usage", response_model=List[Dict])
def get_usage_analytics(period: str = Query("7d", enum=["7d", "30d", "90d"])):
    # Example: Dummy data for usage analytics
    data = [
        {"date": "2026-02-15", "dataset": "users", "views": 120, "unique_users": 30},
        {"date": "2026-02-15", "dataset": "transactions", "views": 80, "unique_users": 22},
        {"date": "2026-02-14", "dataset": "users", "views": 110, "unique_users": 28},
    ]
    return data

# To use: include_router(usage_analytics.router) in main.py

File: src/frontend/src/services/usageAnalytics.js
// [AI GENERATED]
// Model: GitHub Copilot (GPT-4.1)
// Logic: Provides a frontend service stub for fetching data usage analytics from the backend.
// Why: Enables the dashboard and analytics UI to display dataset usage, report popularity, and user engagement.
// Root Cause: No API integration for usage analytics existed.
// Context: Extend this service to support filtering, visualization, and user-specific analytics.
// Model Suitability: For advanced analytics UI, consider GPT-4 Turbo or Claude Sonnet.

import api from './api';

export async function fetchUsageAnalytics(period = '7d') {
    // Fetches usage analytics for the given period (7d, 30d, 90d)
    const res = await api.get('/analytics/usage', { params: { period } });
    return res.data;
}
